### PR TITLE
[TOS-694] fix(test-case-result-filter): label, dates added to filter

### PR DESCRIPTION
**Pull request by nixonakash01**
# Description
* Jira : https://testsigma.atlassian.net/browse/TOS-694
* Steps for reproduce
Create Test Plan
Execute the Test plan
goto the View reports page
Click on Filter icon and observe
* Actual
The Fields such as "Labels , Created date and Updated date" are not displayed under the Filter overlay in the Run Results Page
* Expected
The Fields such as "Labels , Created date and Updated date" to be displayed under the Filter overlay in the Run Results Page
# Fix
added label, created date and updated date in the filter and changed the corresponding condition to disable the filter button

### CODE REVIEW

File: ui/src/app/components/webcomponents/test-case-result-filter.component.html [hunk 1]
```diff
@@ -53,6 +53,53 @@
       </mat-form-field>
       <label class="control-label" [translate]="'testcase.summary.test_case_priority'"></label>
     </div>
+    <div class="form-group">
+      <mat-form-field class="mat-date-custom w-100" appearance="fill">
+        <mat-label [translate]="'testcase.list.created_date_placeholder'"></mat-label>
+        <mat-date-range-input [rangePicker]="createdDateRangePicker" [formGroup]="createdDateRange" [max]="maxDate">
+          <input matStartDate formControlName="start" placeholder="Start date" [max]="maxDate" required>
+          <input matEndDate formControlName="end" placeholder="End date" [max]="maxDate" required>
```
Comment (guganabu) [inline ui/src/app/components/webcomponents/test-case-result-filter.component.html:61]: plz use locales everywhere

File: ui/src/app/components/webcomponents/test-case-result-filter.component.html [hunk 2]
```diff
@@ -53,6 +53,53 @@
...
+           [translate]="'form.validation.common.invalid' | translate:{FieldName: 'Created Date'}" ></div>
+    </div>
+
+    <div class="form-group">
+
+      <mat-form-field class="mat-date-custom w-100" appearance="fill">
+        <mat-label [translate]="'testcase.list.updated_date_placeholder'"></mat-label>
+        <mat-date-range-input [rangePicker]="updatedDateRangePicker" [formGroup]="updatedDateRange" [max]="maxDate">
+          <input matStartDate formControlName="start" placeholder="Start date" [max]="maxDate" required>
+          <input matEndDate formControlName="end" placeholder="End date" [max]="maxDate" required>
```
Comment (guganabu) [inline ui/src/app/components/webcomponents/test-case-result-filter.component.html:78]: use locales

File: ui/src/app/components/webcomponents/test-case-result-filter.component.ts [hunk 3]
```diff
@@ -26,21 +28,41 @@ export class TestCaseResultFilterComponent implements OnInit {
...
+  });
+  maxDate = new Date();
 
   constructor(
     public translate: TranslateService,
     private testCaseTypeService: TestCaseTypesService,
     private testCasePriorityService: TestCasePrioritiesService) {
   }
 
+  dateInvalid(DateRange) {
```
Comment (guganabu) [inline ui/src/app/components/webcomponents/test-case-result-filter.component.ts:49]: it can be named as isDataInvalid to denote conditional check

Comment (nixonakash01) [reply ui/src/app/components/webcomponents/test-case-result-filter.component.ts:49]: okay bro, will change

File: ui/src/app/components/webcomponents/test-case-result-filter.component.html (hunk 2 above)
Comment (nixonakash01) [reply ui/src/app/components/webcomponents/test-case-result-filter.component.html]: pleace check latest commit

File: ui/src/app/components/webcomponents/test-case-result-filter.component.html (hunk 1 above)
Comment (nixonakash01) [reply ui/src/app/components/webcomponents/test-case-result-filter.component.html]: pleace check latest commit

File: ui/src/app/components/webcomponents/test-case-result-filter.component.ts (hunk 3 above)
Comment (guganabu) [reply ui/src/app/components/webcomponents/test-case-result-filter.component.ts]: can we make this isInvalidDateRange which express better actually

**Review by guganabu — APPROVED** [review]

**Pull request merged**